### PR TITLE
Adding a CLI for staticrypt.

### PR DESCRIPTION
**Pull request by AaronCoplan**
This PR adds staticrypt-cli, which is a node module that allows users to run staticrypt on the command line.  Simply install the node module and then run `staticrypt` on the command line to view usage information.  I saw #97 and thought it would be a useful feature to add.

Please let me know of any feedback or changes you'd like made!

### CODE REVIEW

**Comment by robinmoisson** [comment]
Hello @AaronCoplan, thanks for a great PR ! I'm commenting on a few things then I'd love to merge.

**Comment by robinmoisson** [comment]
Sorry, I'm noticing the default repo branch has been switched to `ckeditor` (a WIP branch to help with #21) by mistake a while ago. The main branch should be `gh-pages` so I'm editing the PR to point to the right direction...

There should be only one line impacted in `password_template.html` if we want to keep both `cli` and web version the same ([here](https://github.com/robinmoisson/staticrypt/commit/78597f68ce58112598dc2adff957afd71009e5ac#diff-7e68ff976af37c6606e4a67302f5b640R153), added to help tools other than crypto-js to generate the encrypted string).

**Comment by AaronCoplan** [comment]
@robinmoisson I believe I made all the necessary changes, let me know if there's anything else needed to make it ready to merge!

**Comment by AaronCoplan** [comment]
@robinmoisson Made all changes requested in comments, definitely good points about offline embedding and the naming convention.

File: cli/index.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    encrypted: encryptedMessage,
+    crypto_tag: SCRIPT_TAG,
+    embed: namedArgs.embed != null ? namedArgs.embed : false,
+    outputFilePath: namedArgs.output != null ? namedArgs.output : namedArgs.input.replace('.html', '') + "_encrypted.html"
+};
+
+if(data.embed){
+    try{
+        var embedContents = FileSystem.readFileSync('crypto-js.min.js', 'utf8');
+        data["crypto_tag"] = '<script>' + txt + '</script>';
```
Comment (robinmoisson) [inline cli/index.js]: I believe `txt` is not defined anymore here, it should be `embedContents` :)

File: cli/index.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+var hmac = CryptoJS.HmacSHA256(encrypted.toString(), CryptoJS.SHA256(namedArgs.password).toString()).toString();
+var encryptedMessage = hmac + encrypted;
+
+var data = {
+    title: namedArgs.title != null ? namedArgs.title : "Protected Page",
+    instructions: namedArgs.instructions != null ? namedArgs.instructions : "",
+    encrypted: encryptedMessage,
+    crypto_tag: SCRIPT_TAG,
+    embed: namedArgs.embed != null ? namedArgs.embed : false,
+    outputFilePath: namedArgs.output != null ? namedArgs.output : namedArgs.input.replace('.html', '') + "_encrypted.html"
```
Comment (robinmoisson) [inline cli/index.js]: It's probably never going to pause any issue in the real world but if our filename is `foo.html_bar.html` our final name would be `foo_bar.html_encrypted.html`.

We can use `namedArgs.input.replace(/\.html$/, '')` to only catch `.html` if it's at the extension

**Comment by AaronCoplan** [comment]
@robinmoisson Good catches!  Sorry about that - changes are up

**Comment by robinmoisson** [comment]
Hi @AaronCoplan, I've been doing a few tests and it's looking good, very nice to have a cli. One last thing I think should be changed is the argument parsing: I'm realizing the interface for the command is 

    staticrypt input=test.html password=12345 title="a title"

While for a cli most packages are using a format closer to

    staticrypt <input> <password> [options]
    #ex: 'staticrypt test.html 12345 --title "a title"'

[yargs](https://www.npmjs.com/package/yargs) seems like a good package for parsing the arguments and generating standard-ish doc (http://docopt.org/) without writing to much custom code (see their [examples](https://github.com/yargs/yargs/blob/master/docs/examples.md)).

That would make breaking changes to the interface as it is now, so I think it should be done before the merge. Sorry, I should have told you that sooner - now that the holiday season is up I have much more bandwidth, let me know if you want me to do the changes !

**Comment by AaronCoplan** [comment]
@robinmoisson I have refactored to use yargs and cleaned up a couple other things in the process. I have tested it by running the following sequence of commands:

1. `npm pack`
2. `sudo npm install  -g staticrypt-cli-1.0.0.tgz`
3. `staticrypt <input file> <password>`

This seems to work fine, and I have tested it with various combinations of the options as well.

File: cli/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+const namedArgs = Yargs
+      .usage('staticrypt <input> <password> options')
+      .demandCommand(2)
+      .default({'output': null, 'title': null, 'instructions': null, 'embed': null})
+      .alias('o', 'output')
+      .alias('t', 'title')
+      .alias('i', 'instructions')
+      .alias('e', 'embed')
+      .argv;
```
Comment (robinmoisson) [inline cli/index.js]: If we want to keep the previous help text and show the default values we can use something like:

```
const namedArgs = Yargs
      .usage('Usage: staticrypt <input> <password> [options]')
      .demandCommand(2)
      .option('e', {
        alias: 'embed',
        type: 'boolean',
        describe: 'Whether or not to embed crypto-js in the page (or use an external CDN)',
        default: false
      })
      .option('o', {
        alias: 'output',
        type: 'string',
        describe: 'File name / path for generated encrypted file',
        default: null
      })
      .option('t', {
        alias: 'title',
        type: 'string',
        describe: "Title for output HTML page",
        default: 'Protected Page'
      })
      .option('i', {
        alias: 'instructions',
        type: 'string',
        describe: 'Special instructions to display to the user.',
        default: null
      })
      .argv;
```

File: cli/index.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+if(namedArgs._.length != 2){
+    Yargs.showHelp();
+    process.exit(1);
+}
+
+const input = namedArgs._[0];
+const password = namedArgs._[1];
```
Comment (robinmoisson) [inline cli/index.js]: If my password is `123` it is cast to an int and crypto-js will throw an error, so we need to explicitly use a string:

    const password = namedArgs._[1].toString();

Comment (robinmoisson) [reply cli/index.js]: (for good measure we could do the same for the input)

**Comment by AaronCoplan** [comment]
@robinmoisson Good ideas, I've made the changes and retested, everything seems good to go!

**Comment by robinmoisson** [comment]
Great ! I'm merging, I'll add some info to the readme, revert the extra commit leftover from the wrong starting branch and clean a few things up, then I think it can be published to npm as well. Good work !

**Pull request merged**